paella-matterhorn/plugins/edu.harvard.dce.infoButtonPlugin/dce_infobutton.js

```diff
@@ -51,7 +51,13 @@ Class ('paella.plugins.InfoPlugin', paella.ButtonPlugin,{
       case ('All Course Videos'):
         if (paella.matterhorn && paella.matterhorn.episode && paella.matterhorn.episode.dcIsPartOf){
           var seriesId = paella.matterhorn.episode.dcIsPartOf;
-          location.href = '../ui/publicationListing.shtml?seriesId=' + seriesId;
+          // MATT-1373 reference combo page if valid DCE offering
+          if (seriesId.match('^[0-9]{11}$')) {
+            location.href = '../ui/#/' + seriesId.slice(0,4) + '/' + seriesId.slice(4,6) + '/' + seriesId.slice(6,11);
```

Comment:
If these slices are meaningful, could you extract them to named variable so the next developer understands their intent? Perhaps:

```
var section = seriesId.slice(0,4);
var subSection = seriesId.slice(4,6);
var classId = seriesId.slice(6,11);

# everything else, with the variable being named semantically
```

I'm a big fan of using local variables to both document and make intent clear for future generations.